crates/pessimistic-proof-core/src/local_state.rs

```diff
@@ -90,7 +90,6 @@ impl NetworkState {
             });
         }
 
-        // TODO: benchmark if BTreeMap is the best choice in terms of SP1 cycles
```

Comment:
Note that I have benchmarked independently the switch from `BTreeMap` to `HashMap` (with preallocation) for all 4 sample sizes listed in the PR description, and it didn't bring any improvement (performances actually worsened negligibly).